config/routes.rb

```diff
@@ -7,6 +7,9 @@
   # route GET /login to login controller and #index action
   get "login", to: "login#index"
   # route GET /check-in to login controller and #index action
-  # get "checkin", to: "checkin#new"
+  get "checkin", to: "checkin#new"
   post "checkin", to: "checkin#create"
+  # the admin dashboard
+  get "admins/index"
+  get "admins/view_checkin_records"
```

Comment:
I don't think this design follows rails routing conventions. Since you're trying to view checkins, this should be a GET request to the checkin controller, not the admin controller. Something like `get "checkin", to: "checkin#index" would be better, since you want to view a list of checkins. For more info, take a look at this rails doc page on routing: https://guides.rubyonrails.org/routing.html

Reply:
Just another note, the admin dashboard should just be a hub for admins, not where any of the logic of data gathering or sorting should live. As I see it, it should just be a collection of buttons, links, and tabs which lead you to other routes (like appointments, checkins, students, etc.) so in this sense it's mainly just a view. Let me know if that makes sense or not

Reply:
I make `view_checkin_records` as an admins_controller method because only admins have permission to view the check-in records. If we put `view_checkin_records` in checkin controller, then we need to check if the user is an admin in checkin controller and handle the case if the user is not an admin but tries to access the page. Checking if the user is an admin is a private method I want to put in admin_controller, so I think it is better to put `view_checkin_records` in admins_controller since putting `view_checkin_records` in checkin_controller is not DRY.